### DISCUSSION TITLE
WPF - Drag an attachment from outlook

### DISCUSSION THREAD

**Discussion by ghost** [cefsharp/CefSharp#3321] [General]
Can anyone give me some help with dragging an attachment from outlook into a web page hosted in CefSharp WPF. I understand that chromium has a fx for this but it still has to be implemented in CefSharp? Can anyone give some tips and I can try and do it in my own branch. Thanks.

**Comment by amaitland** [comment]
Have you enabled drag and drop on the ChromiumWebBrowser? 

https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.allowdrop?view=net-5.0

As a starting point I'd add a breakpoint to
https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Wpf/ChromiumWebBrowser.cs#L1575

The CEF API https://github.com/chromiumembedded/cef/blob/4f28c5ffffd00f9a8fa67392e594e5207cb65158/include/cef_drag_data.h#L51

**Reply by ghost** [reply]
Hello,

AllowDrop is enabled on the CefSharp browser and drop operations are working from files.

We have a specific problem with dragging an outlook attachment to an embedded web page.

This is solved in the new WebView2 control so we are switching to embedded edge via this control.

Thanks

**Reply by amaitland** [reply]
From the little I know about the `WebView2` implementation in `WPF` it's not actually a native `WPF` control, it's a `Win32` control (basically like hosting `WinForms` in `WPF`).

For what it's worth the [CefSharp.Wpf.HwndHost](https://www.nuget.org/packages/CefSharp.Wpf.HwndHost/) package is indented to be a drop in replacement for those wishing to host the `Win32` version in `WPF`. You will have all the same airspace issues.

**Reply by mukeshmishra996** [reply]
Hi @amaitland  - There is any update on drag and drop from Outlook. I have also similar kind of requirement. Can you please help me to achieve this requirement.
